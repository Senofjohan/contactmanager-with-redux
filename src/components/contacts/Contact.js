import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { deleteContact, clearContacts } from '../../actions/contactActions';

class Contact extends Component {
  constructor(props) {
    super(props);
    this.onClearClick = this.onClearClick.bind(this);
  }
  state = {
    showContactInfo: false
  };

  onDeleteClick = id => {
    this.props.deleteContact(id);
  };

  onClearClick = () => {
    if (window.confirm('Are you sure you want to delete ALL contacts?')) {
      this.props.clearContacts();
    }
  };

  render() {
    const { id, name, email, phone } = this.props.contact;
    const { showContactInfo } = this.state;

    return (
      <div className="card card-body mb-3">
        <h4>
          {name}{' '}
          <i
            onClick={() =>
              this.setState({
                showContactInfo: !this.state.showContactInfo
              })
            }
            className="fas fa-sort-down"
            style={{ cursor: 'pointer' }}
          />
          <i
            title="Delete"
            className="fas fa-times"
            style={{ cursor: 'pointer', float: 'right', color: 'red' }}
            onClick={this.onDeleteClick.bind(this, id)}
          />
          <Link
            title="Edit"
            aria-label={`edit contact ${name}`}
            to={`contact/edit/${id}`}
          >
            <i
              className="fas fa-pencil-alt"
              style={{
                cursor: 'pointer',
                float: 'right',
                color: 'black',
                marginRight: '1rem'
              }}
            />
          </Link>
          <i
            title="Clear all contacts"
            className="fas fa-trash"
            style={{
              cursor: 'pointer',
              float: 'right',
              color: 'red',
              marginRight: '1rem'
            }}
            // onClick={() => {
            //   if (window.confirm('Are you sure you wish to clear all contacts'))
            //     this.onClearClick.bind(this);
            // }}
            onClick={this.onClearClick}
          />
        </h4>
        {showContactInfo ? (
          <ul className="list-group">
            <li className="list-group-item">Email: {email}</li>
            <li className="list-group-item">Phone: {phone}</li>
          </ul>
        ) : null}
      </div>
    );
  }
}

Contact.propTypes = {
  contact: PropTypes.object.isRequired,
  deleteContact: PropTypes.func.isRequired,
  clearContacts: PropTypes.func.isRequired
};
// calling delete and so don't need to mapStateToProps so passing in null
export default connect(
  null,
  { deleteContact, clearContacts }
)(Contact);
