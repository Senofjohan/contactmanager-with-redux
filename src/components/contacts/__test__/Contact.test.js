import Contact from '../Contact';
import uuid from 'uuid';
// app/src/components/__tests__/Login-test.js
import configureStore from 'redux-mock-store';

describe('<Contact />', () => {
  let wrapper;
  let store;
  beforeEach(() => {
    const initialState = {};
    // here it is possible to pass in any middleware if needed into //configureStore
    const mockStore = configureStore();
    //creates the store with any initial state or middleware needed
    store = mockStore(initialState);
  });

  it('renders correctly ', () => {
    wrapper = shallow(
      <Contact
        store={store}
        key={uuid()}
        contact={{
          id: 1,
          name: 'John Doe',
          email: 'john@gmail.com',
          phone: '555-555-5555'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
