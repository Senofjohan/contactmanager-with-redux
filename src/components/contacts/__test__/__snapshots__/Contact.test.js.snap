// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders correctly  1`] = `
<Contact
  clearContacts={[Function]}
  contact={
    Object {
      "email": "john@gmail.com",
      "id": 1,
      "name": "John Doe",
      "phone": "555-555-5555",
    }
  }
  deleteContact={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<Home /> renders correctly  1`] = `
<Contact
  clearContacts={[Function]}
  contact={
    Object {
      "email": "john@gmail.com",
      "id": 1,
      "name": "John Doe",
      "phone": "555-555-5555",
    }
  }
  deleteContact={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`renders correctly  1`] = `
<Contact
  clearContacts={[Function]}
  contact={
    Object {
      "email": "john@gmail.com",
      "id": 1,
      "name": "John Doe",
      "phone": "555-555-5555",
    }
  }
  deleteContact={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
